### ISSUE TITLE
Update hera_cal to use integration times as arrays 

### ISSUE THREAD

**Issue by jsdillon** [HERA-Team/hera_cal#348]
In order to get up to date with master on pyuvdata, we need to fix our handling of integration times. A few places where I think this is broken:

* https://github.com/HERA-Team/hera_cal/blob/8044e56eb4297eb589c1757eb4a5ef4a976dafb4/hera_cal/cal_formats.py#L121

* https://github.com/HERA-Team/hera_cal/blob/8044e56eb4297eb589c1757eb4a5ef4a976dafb4/hera_cal/io.py#L703

* https://github.com/HERA-Team/hera_cal/blob/c6e563e823d71cdc0b5f0b5baa15bd54438170c0/hera_cal/abscal_funcs.py#L1727

This currently open PR fixes issues with current uvh5 test files:
https://github.com/RadioAstronomySoftwareGroup/pyuvdata/pull/430

This is all made a bit trickier because, as you noted @nkern,
```    
integration_time : type=float, integration duration in seconds for data_array. This does not necessarily have
        to be equal to the diff(time_array): for the case of LST-binning, this is not the duration of the LST-bin
        but the integration time of the pre-binned data.
```

This change in pyuvdata definitely changes your assumptions, so I'm assigning you to fix it. I think we probably want to stop using integration times of pre-binned data and instead increase the integration times and decrease nsamples accordingly. I'm not sure how this propagates through to hera_pspec though...

When you're done, please undo the change I made here so we're testing against master again: https://github.com/HERA-Team/hera_cal/pull/347/commits/f32fb8b6e822e89fc94de98886c8731d5d99e3ff

**Comment by jsdillon** [comment]
Since you've been looking at this, any other thoughts @plaplant?

**Comment by jsdillon** [comment]
Resolved by #360.
